File: CrowdCheer/RunnerAnnotation.m
//
//  RunnerAnnotation.m
//  CrowdCheer
//
//  Created by Leesha Maliakal on 8/24/15.
//  Copyright (c) 2015 Delta Lab. All rights reserved.
//

#import "RunnerAnnotation.h"

@implementation RunnerAnnotation

-(id)initWithTitle:(NSString *)newTitle Location:(CLLocationCoordinate2D)location RunnerID:(NSString *)runnerID
{
    self = [super init];
    if(self)
    {
        _title = newTitle;
        _coordinate = location;
        _runnerObjID = runnerID;
    }
    return self;
}

-(MKAnnotationView *)annotationView
{
    UIButton *cheerBtn = [UIButton buttonWithType:UIButtonTypeContactAdd];
    [cheerBtn setTitle:@"Cheer!" forState:UIControlStateNormal];
    
    MKAnnotationView *annotationView = [[MKAnnotationView alloc]initWithAnnotation:self reuseIdentifier:@"RunnerAnnotation"];
    annotationView.enabled = YES;
    annotationView.canShowCallout = YES;
    annotationView.image = [UIImage imageNamed:@"runner.png"];
    annotationView.rightCalloutAccessoryView = cheerBtn;
    
    return annotationView;
}

@end

File: CrowdCheer/RunnerAnnotation.swift
//
//  RunnerAnnotation.swift
//  CrowdCheer
//
//  Created by Leesha Maliakal on 2/10/16.
//  Copyright © 2016 Delta Lab. All rights reserved.
//

import Foundation
import UIKit
import MapKit

enum RunnerType: Int {
    case RunnerDefault = 0
    case MyRunner
}

class TrackRunnerAnnotation: NSObject, MKAnnotation {
    var coordinate: CLLocationCoordinate2D
    var title: String?
    var subtitle: String?
    var type: RunnerType
    var image: UIImage
    
    init(coordinate: CLLocationCoordinate2D, title: String, subtitle: String, type: RunnerType, image: UIImage) {
        self.coordinate = coordinate
        self.title = title
        self.subtitle = subtitle
        self.type = type
        self.image = image
    }
}


class PickRunnerAnnotation: NSObject, MKAnnotation {
    var coordinate: CLLocationCoordinate2D
    var title: String?
    var type: RunnerType
    var runnerObjID: String?
    
    init(coordinate: CLLocationCoordinate2D, title: String, type: RunnerType, runnerObjID: String) {
        self.coordinate = coordinate
        self.title = title
        self.type = type
        self.runnerObjID = runnerObjID
    }
}